backend/services/esweService.ts

```diff
@@ -162,16 +164,74 @@ export default class EsweService {
     return createFlaggedContent(content)
   }
 
-  async getLearnerEmployabilitySkills(nomisId: string): Promise<LearnerEmployabilitySkills> {
-    let result: LearnerEmployabilitySkills = null
+  async getLearnerEmployabilitySkills(nomisId: string): Promise<EmployabilitySkills> {
+    const map = new Map<EmployabilitySkill, EmployabilitySkillsReview>()
     try {
       const context = await this.systemOauthClient.getClientCredentialsTokens()
-      result = await this.curiousApi.getLearnerEmployabilitySkills(context, nomisId)
+      const result = await this.curiousApi.getLearnerEmployabilitySkills(context, nomisId)
+
+      return createFlaggedContent(
+        result.content.reduce((acc, current) => {
+          if (current.reviews?.length) {
+            const reviewLatest = current.reviews.reduce(function (arr, r) {
+              return arr.reviewDate > r.reviewDate ? arr : r // ISO dates so can compare lke strings
+            })
+            const latest = acc.get(current.employabilitySkill)
+            if (latest) {
+              if (reviewLatest?.reviewDate && reviewLatest.reviewDate > latest.reviewDate) {
+                acc.set(current.employabilitySkill, reviewLatest)
+              }
+            } else if (reviewLatest?.reviewDate) {
+              acc.set(current.employabilitySkill, reviewLatest)
+            }
+          }
+          return acc
+        }, map)
+      )
+    } catch (e) {
+      if (e.response?.status === 404) {
+        log.info(`Offender record not found in getLearnerEmployabilitySkills().`)
+        return createFlaggedContent(map)
+      }
+      log.error(`Failed in getLearnerEmployabilitySkills(). Reason: ${e.message}`)
+    }
+
+    return createFlaggedContent(null)
+  }
+
+  getOrPut = (map, key, def) => {
+    const value = map.get(key)
+    if (value) {
+      return value
+    }
+    map.set(key, def)
+    return def
+  }
+
+  async getLearnerEmployabilitySkillsDetails(nomisId: string): Promise<EmployabilitySkillsDetails> {
+    const map = new Map<EmployabilitySkill, EmployabilitySkillsReview[]>()
+    try {
+      const context = await this.systemOauthClient.getClientCredentialsTokens()
+      const result = await this.curiousApi.getLearnerEmployabilitySkills(context, nomisId)
+
+      return createFlaggedContent(
+        result.content.reduce((acc, current) => {
```

Comment:
I think this `reduce` code block would benefit from been in it's own function with a name that indicates what it does.

Again, one for later.